core/src/main/java/io/jenkins/pluginhealth/scoring/probes/JSR305Probe.java

```diff
@@ -0,0 +1,43 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.util.List;
+
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+
+/**
+ * This probe checks for deprecated @Nonnull and @CheckForNull annotations in a plugin.
+ */
+@Component
+@Order(value = AbstractDeprecatedAnnotationImportsCheckProbe.ORDER)
+public class JSR305Probe extends AbstractDeprecatedAnnotationImportsCheckProbe {
```

Comment:
Here I was not able to think a good name for the probe. So I went ahead with this one.

Reply:
lgtm